### PR TITLE
Deduplicate and unify VF2 layout passes

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

In #7862 we recently added a new vf2 post layout pass which is designed
to run after routing to improve the layout once we know there is at
least one isomorphic subgraph in the coupling graph for the interactions
in the circuit. In that PR we still ran vf2 post layout even if vf2
layout pass found a match. That's because the heuristic scoring of
layouts used for the vf2 layout and vf2 post layout passes were
different. Originally this difference was due to the the vf2 post
layout pass being intedended to run after the optimization loop where we
could guarantee the gates were in the target and exactly score the error
for each potential layout. But since the vf2 post layout was updated to
score a layout based on the gate counts for each qubit and the average
1q and 2q instruction error rates we can leverage this better heuristic
scoring in the vf2 layout pass. This commit updates the vf2 layout pass
to use the same heuristic and deduplicates some of the code between the
passes at the same time. Additionally, since the scoring heuristics are
the same the preset pass managers are updated to only run vf2 post
layout if vf2 layout didn't find a match. If vf2 layout finds a match
it's going to be the same as what vf2 post layout finds so there is no
need to run the vf2 post layout pass anymore.

### Details and comments

### CODE REVIEW

File: qiskit/transpiler/preset_passmanagers/level1.py [hunk 1]
```diff
@@ -307,7 +307,17 @@ def _opt_control(property_set):
...
+                if _trivial_not_perfect(property_set):
+                    vf2_stop_reason = property_set["VF2Layout_stop_reason"]
+                    if (
+                        vf2_stop_reason is None
+                        or vf2_stop_reason != VF2LayoutStopReason.SOLUTION_FOUND
+                    ):
+                        return True
+                return False
+
+            def _apply_post_layout_condition(property_set):
```
Comment (kdk) [inline qiskit/transpiler/preset_passmanagers/level1.py:320]: The comment on the lines below looks to be outdated. Can you update it?

File: qiskit/transpiler/preset_passmanagers/level1.py [hunk 2]
```diff
@@ -307,7 +307,17 @@ def _opt_control(property_set):
             and pass_manager_config.layout_method is None
         ):
 
-            def _post_layout_condition(property_set):
+            def _run_post_layout_condition(property_set):
+                if _trivial_not_perfect(property_set):
+                    vf2_stop_reason = property_set["VF2Layout_stop_reason"]
+                    if (
+                        vf2_stop_reason is None
+                        or vf2_stop_reason != VF2LayoutStopReason.SOLUTION_FOUND
```
Comment (kdk) [inline qiskit/transpiler/preset_passmanagers/level1.py:315]: Is there a reason to prefer comparing by equality here (`!=`) and identity (`is`) below on L326?

**Review by kdk — COMMENTED** [review]
Two small comments, otherwise looks good to me.

File: qiskit/transpiler/preset_passmanagers/level1.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/transpiler/preset_passmanagers/level1.py:315]: Not really, besides being sloppy for enums either works: https://docs.python.org/3/library/enum.html#comparisons

File: qiskit/transpiler/preset_passmanagers/level1.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/transpiler/preset_passmanagers/level1.py:320]: Updated in: https://github.com/Qiskit/qiskit-terra/pull/7991/commits/96e4d281ad2feb75cd1844e82349c195342f42b3

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
I think `qiskit.transpiler.passes.layout.vf2_layout.VF2Layout._score_layout` also needs to be removed as part of the duplication. Not fully sure.

**Comment by mtreinish** [comment]
> I think `qiskit.transpiler.passes.layout.vf2_layout.VF2Layout._score_layout` also needs to be removed as part of the duplication. Not fully sure.

Good catch I meant to remove it but forgot. Done in: https://github.com/Qiskit/qiskit-terra/pull/7991/commits/ff875e85ae5499d1d1aefc2dc8f8c28420b0ae75

**Review by 1ucian0 — APPROVED** [review]
Let's get this merged!

**Pull request merged**